addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/LEDStateDTO.java

```diff
@@ -70,18 +75,18 @@ public static LEDStateDTO valueOf(int state, int program, int programSpeed, int
         OnOffType power = (state & 0x01) != 0 ? OnOffType.ON : OnOffType.OFF;
         float[] hsv = new float[3];
         Color.RGBtoHSB(red, green, blue, hsv);
-        long hue = (long) (hsv[0] * 360);
-        int saturation = (int) (hsv[1] * 100);
-        int brightness = (int) (hsv[2] * 100);
+        BigDecimal hue = BigDecimal.valueOf(hsv[0] * 360);
```

Comment:
I agree with both of you.  `RGBtoHSB(...)` and the multiplication `hsv[0] * 360` are done in float.  So it doesn't make a difference regarding the precision if `hue` is declared as `float` or `BigDecimal`.  But since `hue` is anyway converted to `BigDecimal` in line 81 (in `DecimalType`), there is also no difference in the performance.  So `float` and `BigDecimal` are equally fine and it's only a matter of code style which to use.  IMO the only sad thing is that there is no version of Color.RGBtoHSB that uses double values instead of float.  Float is pretty much deprecated today, since float and double arrays consume the size of memory in 64 bit JVMs and the float and double operations have even the same performance in 64 bit JVMs.  (At least that was the case when I made some performance tests with Oracle Java several years ago.)